src/otx/v2/adapters/torch/mmengine/mmaction/modules/datasets/pipelines/loading.py

```diff
@@ -0,0 +1,49 @@
+"""Collection of video loading data pipelines for OTX Action Task."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from mmaction.registry import TRANSFORMS
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset import Dataset as DatumDataset
+
+@TRANSFORMS.register_module(force=True)
+class OTXRawFrameDecode:
+    """Load and decode frames with given indices."""
+
+    otx_dataset: DatumDataset
+
+    def __call__(self, results: dict[str, Any]) -> dict[str, Any]:
+        """Call function of RawFrameDecode."""
+        return self._decode_from_list(results)
+
+    def _decode_from_list(self, results: dict[str, Any]) -> dict:
+        """Generate numpy array list from list of DatasetItemEntity."""
+        imgs = []
+        for index in results["frame_inds"]:
+            item_id = results["item_ids"][index]
```

Comment:
Quick question, are we going to parse all items with the same `frame_inds`?
For example, when video1 contains frame_inds 3 and video 2 also contains frame_inds3, then we are going to append all images with the frame_inds from different videos?

Reply:
Before calling the `OTXRawFrameDecode`, there was a `SampleFrames` pipeline. It packs each `results` with individual video id. So, in this case, `results` only means for the single video.

(i.e. All item_ids have the `1##000**` that means it comes from the video_1)
<img width="974" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/80735344/794d8b92-fda8-44dc-a960-f79185d2ace0">